testplan/testing/multitest/base.py

```diff
@@ -547,6 +549,18 @@ def create_testcase_report(testcase):
         if not testsuite_report.failed:
             testsuite_report.status = Status.PASSED
 
+        # If xfailed testsuite, force set status_override and update result
+        if testsuite_report.xfail:
```

Comment:
At this point, instead of checking the report xfail, I think you should instead check the testcase's `__xfail__` attribute - if it is `True` and the testcase report status is `Status.FAIL`, update it to `Status.XFAIL`.

I'm not sure if possible but it would be better if you can directly update the status instead of using the status_override attribute.